src/include/containers/List.h

```diff
@@ -187,7 +187,9 @@ void List<ValueType>::clear() {
 
 template <typename ValueType>
 void List<ValueType>::insert(int index, ValueType value) {
-	if (index != 0) boundaryCheck(index);
+	if (index < 0) {
```

Comment:
But the stl list allows that... if you use `std::advance(itr, x)` where x beyond range, even though it's an undefined behavior, but it seems that the itr will be `end()`